audobject/core/api.py

```diff
@@ -16,14 +19,14 @@
 )
 
 
-def from_dict(  # noqa: D417
-    d: typing.Dict[str, typing.Any],
-    root: str = None,
+def from_dict(
+    d: Mapping[str, object],
+    root: str | None = None,
```

Comment:
In the other packages, for which we updated the syntax, we have not added `None` as type, if it was given already as default value. Is it on purpose that you decided to explicitly write it here?

Reply:
Ok this is too explicit then. Logically it is ok: when you have a default value of `None` then it must present a valid input type. 

I have refactored such that the `None` is maintained whenever 
there had been a `typing.Optional`  before.  see cb4fc2c

Reply:
Thanks